Src/ILGPU/IndexTypes.tt

```diff
@@ -633,6 +622,22 @@ namespace ILGPU
             <#= name #> dimension) =>
             linearIndex;
 
+<# if (def.IsIntIndex) { #>
+        /// <summary>
+        /// Reconstructs a 1D index from a linear index.
+        /// </summary>
+        /// <param name="linearIndex">The linear index.</param>
+        /// <param name="dimension">The 1D dimension for reconstruction.</param>
+        /// <returns>The reconstructed 1D index.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static <#= name #> ReconstructIndex(
+            long linearIndex,
```

Comment:
Some tests would be appropriate, I think, for all of these guys. Should I put them in a new file, Indices.cs, or in ArrayViews.cs?

Reply:
This might also be worth investigating. Related to the question of whether we should write additional tests for each of them, is the discussion in #346, I will comment on in the next few days. I had an offline discussion with several people from the web and the community about possible ways to implement a significantly better API that could affect the current implementation of all `ArrayViewXD`, `IndexXD` and `LongIndexXX` types. This also raises the question on how much effort we should put into the tests discussed above.

Reply:
I added a few simple tests, just for IndexXD, that I think are better than nothing.